### DISCUSSION TITLE
How do I integrate vcpkg to my project in github actions

### DISCUSSION THREAD

**Discussion by xarthurx** [microsoft/vcpkg#19073] [Ideas]
Please see the detailed scenario here:
https://github.com/actions/virtual-environments/issues/3689

Simply put, I'm trying to use `vcpkg` in a github action to compile my library.
However, whatever works locally does not work on the github virtual machine.

It seems that using `msbuild` to build my lib on the virtual machine does not recognize `vcpkg` library location even I run 

`vcpkg integrate install`

Any clue what is the problem?
May it be that my `.sln` file is generated on my local machine and only point to a `vcpkg` location defined locally?
Is it that the `vcpkg` library location is not defined by global environmental variable?

Thanks.

**Comment by JackBoosY** [comment]
I suggest @BillyONeal to have a reply for this issue.

**Comment by BillyONeal** [comment]
`integrate install` relies on a feature in the public msbuild targets that go through `%APPDATA%` which might not work from a service account (as Actions is likely to be using). I think we might need a feature that works without such global integration to enable this scenario....

**Reply by dsame** [reply]
@BillyONeal can you please clarify the meaning of "relies on a feature in the public msbuild targets that go through %APPDATA%"?

Actually the problems with github actions seems to be caused by the installation dir of vcpkg is `C:\vcpkg`. As a result we need to point `vcpkg integrate install` to this directory. Is there a known way to do this?

**Reply by BillyONeal** [reply]
>@BillyONeal can you please clarify the meaning of "relies on a feature in the public msbuild targets that go through %APPDATA%"?

If I understand the link correctly it isn't using the standard public msbuild targets, it's using some "metaproj" thing that vcpkg doesn't understand how to hook in to necessarily.

>Actually the problems with github actions seems to be caused by the installation dir of vcpkg is C:\vcpkg. As a result we need to point vcpkg integrate install to this directory. Is there a known way to do this?

If this is about a copy of vcpkg that the `actions` VMs have on them, you will need to use `integrate install` on that vcpkg instance, not on some other instance. I observe that the output there indicates that it *is* trying to use C:\vcpkg:

>VcpkgTripletSelection:
> Using triplet "x64-windows" from "**C:\vcpkg\scripts\buildsystems\msbuild**\..\..\..\installed\x64-windows\"

Perhaps you installed x86-windows but your project is choosing x64-windows?

**Comment by xarthurx** [comment]
Any updates on this issue?
We did some experiment: https://github.com/actions/virtual-environments/issues/3689

Running a VS project on a different machine also doesn't not work. Is the reason that a VS project contains info of the `integrate install` of the machine where the project is created?
